lib/util/package-rules/package-prefixes.ts

```diff
@@ -34,16 +34,31 @@ export class PackagePrefixesMatcher extends Matcher {
   }
 
   override excludes(
-    { depName }: PackageRuleInputConfig,
-    { excludePackagePrefixes }: PackageRule,
+    { depName, packageName }: PackageRuleInputConfig,
+    packageRule: PackageRule,
   ): boolean | null {
+    const { excludePackagePrefixes } = packageRule;
     if (is.undefined(excludePackagePrefixes)) {
       return null;
     }
     if (is.undefined(depName)) {
       return false;
     }
 
-    return excludePackagePrefixes.some((prefix) => depName.startsWith(prefix));
+    if (
+      is.string(packageName) &&
+      excludePackagePrefixes.some((prefix) => packageName.startsWith(prefix))
+    ) {
+      return true;
+    }
+    if (excludePackagePrefixes.some((prefix) => depName.startsWith(prefix))) {
+      logger.once.info(
+        { packageName, depName },
+        'Use excludeDepPatterns instead of excludePackagePrefixes',
```

Comment:
we should add `excludeDepPrefixes` matcher

Reply:
https://github.com/renovatebot/renovate/issues/27048